jujugui/static/gui/src/app/app.js

```diff
@@ -1934,11 +1934,17 @@ YUI.add('juju-gui', function(Y) {
       // Clear uncommitted state.
       this.env.get('ecs').clear();
       // Disconnect and reconnect the environment.
-      this.env.ws.onclose = function(event) {
-        this.env.on_close();
-        this.env.connect();
-      }.bind(this);
-      this.env.close();
+      var env = this.env;
+      var onclose = function() {
+        env.on_close();
+        env.connect();
+      };
+      if(this.env.ws) {
```

Comment:
We only need it in onclose since "this" can alter where it will be called and I wasn't sure how best to bind it in that instance. I can use env everywhere if that seems clearer to you.